src/transforms/rotation.jl

```diff
@@ -0,0 +1,70 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    Rotation(rot)
+
+Rotate geometry or mesh with rotation `rot` from ReferenceFrameRotations.jl. 
+
+## Examples
+
+```julia
+Rotation(ReferenceFrameRotations.EulerAngleAxis(pi/4, [1, 0, 0]))
+```
+
+"""
+struct Rotation{R} <: GeometricTransform
+  rot::R
+end
+
+"""
+    Rotation(axis, α; degrees=true)
+
+Perform the rotation of angle `α` around the axis directed by the 
+vector `axis`. The angle must be given in degrees if `degrees=true`, 
+otherwise in radians.
+
+"""
+function Rotation(axis::Vector{Real}, α::Real; degrees::Bool = true)
+  axisNorm = norm2(axis)
+  if axisNorm == 0
+    error("The `axis` vector is null.")
+  end
+  v = axis / axisNorm
+  if degrees
+    α = α * pi / 180
+  end
+  Rotation(EulerAngleAxis(α, v))
+end
+
+isrevertible(::Type{<:Rotation}) = true
+
+function preprocess(transform::Rotation, object)
+  rot = transform.rot
+  local M, M⁻¹
+  if typeof(rot) <: DCM
+    M = rot
+    M⁻¹ = inv_rotation(M)
+  else
+    M = convert(DCM, rot)
+    M⁻¹ = convert(DCM, inv(rot))
+  end
```

Comment:
Notice that you tend to split your reasoning into different types. That is rarely needed in generic code. You can simply perform a single operation assuming that all types of rotation follow a consistent api:

```suggestion
  R, R⁻¹ = rot, inv_rotation(rot)
  M, M⁻¹ = convert.(DCM, (R, R⁻¹))
```

Reply:
No! `inv_rotation` works only for `DCM`. For the other types, one has to use `inv`. Sorry I was not clear.

Reply:
Well, then we can replace inv_rotation by inv and everything should work right? You first call the correct inv method for the rotation type and then convert to DCM (matrix) representation for matrix multiplication. I think ReferenceFrameRotations.jl doesn't even need to convert the object to matrix to perform rotation. As far as I remember you can multiply rotation objects directly by vectors.

Reply:
@stla can you please double check this issue? I think you don't need to convert to DCM in order to apply the rotation. The ReferenceFrameRotations.jl package allows the application of the rotation with any rotation type.

Reply:
Was wrong. The `inv_rotation` function is generic, it works for all types of rotations. 
I already tried to multiply a rotation object by a vector and that didn't work. And I just checked the docs, there's no multiplication method for that.